cmd/setup-etcd-environment/run.go

```diff
@@ -36,22 +39,78 @@ var (
 		Long:  "",
 		RunE:  runRunCmd,
 	}
-
-	runOpts struct {
-		discoverySRV string
-		ifName       string
-		outputFile   string
-		bootstrapSRV bool
-	}
+	runOpts opts
 )
 
+type opts struct {
+	discoverySRV string
+	ifName       string
+	outputFile   string
+	bootstrapSRV bool
+}
+
+type SetupEnv struct {
+	opts            *opts
+	etcdDNS         string
+	etcdName        string
+	etcdDataDir     string
+	etcdIP          string
+	Client          *kubernetes.Clientset
+	unsupportedArch string
+}
+
 func init() {
 	rootCmd.AddCommand(runCmd)
 	rootCmd.PersistentFlags().StringVar(&runOpts.discoverySRV, "discovery-srv", "", "DNS domain used to populate envs from SRV query.")
 	rootCmd.PersistentFlags().StringVar(&runOpts.outputFile, "output-file", "", "file where the envs are written. If empty, prints to Stdout.")
 	rootCmd.PersistentFlags().BoolVar(&runOpts.bootstrapSRV, "bootstrap-srv", true, "use SRV discovery for bootstraping etcd cluster.")
 }
 
+func NewSetupEnv(runOpts *opts, etcdName, etcdDataDir string, ips []string) (*SetupEnv, error) {
```

Comment:
minor nit: does this need to be a public function? (this is a self-contained executable, so it perhaps doesn't matter!